back/app/models/user.rb

```diff
@@ -461,6 +461,7 @@ def reset_confirmation_with_no_password
       self.email_confirmation_code_reset_count = 0
     end
     self.confirmation_required = true
+    self.email_confirmation_code_sent_at = nil
```

Comment:
Added this as it should get set when an email is sent, so gives us a better indication of when another email has been sent or not.